CIP-0029/CIP-0029.md

```diff
@@ -0,0 +1,103 @@
+---
+CIP: 29
+Title: Phase-1 Monetary Scripts Serialization Formats
+Authors: Matthias Benkort <matthias.benkort@iohk.io>
+Comments-URI: https://github.com/cardano-foundation/CIPs/pulls/117
+Status: Draft
+Type: Standards
+Created: 2020-08-17
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This specification describes how to serialize Phase-1 monetary scripts (a.k.a. _"native scripts"_) to various formats (JSON, CBOR) to facilitate inter-operability between applications. 
+
+# Motivation
+
+While the existence of scripts is well-known, there's no agreed upon format when it comes to how those scripts shall be shared across applications. Plus, the on-chain representation of such script isn't really documented anywhere at the moment which makes it challenging for DApps to use them. 
+
+# Specification
+
+This specification covers at present two serialization formats: JSON and CBOR. The CBOR matches exactly the on-chain representation and is extracted from the cardano-ledger-specs source code and put here as a convenient place to lookup while the source can change location over time. It also covers the _full serialization_ (with serialization tag) which is missing from the CDDL specification.
```

Comment:
I don't know what this "full serialisation" refers to. This is covered by the CDDL and there is no extra tag for scripts within a tx (in a witness set).

The prefix is only used for hashing.

Reply:
Yes indeed, wording is poorly chosen here and clarified at the end. This refers to the extra tag needed for the hash preimage. And you're right, it's in the cddl now! I missed that!